grakn-engine/src/main/java/ai/grakn/engine/controller/TasksController.java

```diff
@@ -165,54 +188,190 @@ private Json createTask(Request request, Response response) {
+            @ApiImplicitParam(name = REST.Request.CONFIGURATION_PARAM, value = "JSON Object that will be given to the task as configuration.", dataType = "String", paramType = "body")
+    })
+    private Json createTaskBulk(Request request, Response response) {
+        Json requestBodyAsJson = bodyAsJson(request).at("value");
+        Json configuration = extractConfiguration(requestBodyAsJson);
+        List<Json> taskJsonList = requestBodyAsJson.at(REST.Request.TASKS_PARAM).asJsonList();
+        Json responseJson = Json.array();
+        response.type(ContentType.APPLICATION_JSON.getMimeType());
+        // We need to return the list of taskStates in order
+        // so the client can relate the state to each element in the request.
+        List<TaskStateWithIndex> taskStates = new ArrayList<>();
+        for (int i = 0; i < taskJsonList.size(); i++) {
+            Json singleTaskJson = taskJsonList.get(i);
+            try {
+                taskStates
+                        .add(new TaskStateWithIndex(extractParametersAndProcessTask(singleTaskJson),
+                                i));
+            } catch (Exception e) {
+                LOG.error("Malformed request at {}", singleTaskJson, e);
+                // We return a failure for the full request as this imply there is
+                // something wrong in the client logic that needs to be addressed
+                response.status(HttpStatus.SC_BAD_REQUEST);
+                return Json.object();
+            }
+        }
+
+        List<CompletableFuture<Json>> futures = taskStates.stream()
+                .map(taskState -> CompletableFuture
+                        .supplyAsync(() -> addTaskToManager(configuration, taskState), executor))
+                .collect(toList());
+        CompletableFuture<List<Json>> completableFuture = all(futures);
+        try {
+            List<Json> results = completableFuture
+                    .get(MAX_EXECUTION_TIME.getSeconds(), TimeUnit.SECONDS);
+            boolean hasFailures = false;
+            for (Json resultForTask : results) {
+                responseJson.add(resultForTask);
+                if (resultForTask.at("code").asInteger() != HttpStatus.SC_OK) {
+                    hasFailures = true;
+                }
+            }
+            if (!hasFailures) {
+                response.status(HttpStatus.SC_OK);
+            } else if (responseJson.asJsonList().size() > 0) {
+                response.status(HttpStatus.SC_ACCEPTED);
+            } else {
+                response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+            }
+            return responseJson;
+        } catch (TimeoutException | InterruptedException e) {
+            LOG.error("Task interrupted", e);
+            response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Im not sure if this is standard, but it might be nice if we included a bit of the exception when we send the response back to the client so they know what went wrong.